lib/pengine/complex.c

```diff
@@ -873,10 +872,10 @@ pe__unpack_resource(xmlNode *xml_obj, pe_resource_t **rsc,
 
     if (expanded_xml) {
         if (add_template_rsc(xml_obj, data_set) == FALSE) {
-            return FALSE;
+            return pcmk_rc_unpack_error;
```

Comment:
`add_template_rsc()`'s direct failure conditions would be `pcmk_rc_schema_validation`, but it calls `add_tag_ref()`, which returns `FALSE` for memory allocation failure.

Reply:
Yep, ditto for add_template_rsc()'s return value. When we know something is a memory error, ENOMEM is appropriate, but when a memory error is mixed in with other possibilities, it's fine to ignore the memory case and focus on the rest, because a memory issue will very quickly result in a fatal exit and/or fencing, and it might not even be possible to log a message.

Reply:
Ah, fair enough.